test/test_smoke.py

```diff
@@ -11,39 +12,97 @@
 from vm import VM
 
 
-@pytest.fixture(name="output_path")
-def output_path_fixture(tmp_path):
-    output_path = tmp_path / "output"
-    output_path.mkdir(exist_ok=True)
-    return output_path
+if not testutil.has_executable("podman"):
+    pytest.skip("no podman, skipping integration tests that required podman", allow_module_level=True)
+
+if os.getuid() != 0:
+    pytest.skip("tests require root to run", allow_module_level=True)
+
+# building an ELN image needs x86_64-v3 to work, we use avx2 as a proxy
+# to detect if we have x86-64-v3 (not perfect but should be good enough)
+if not testutil.has_x86_64_v3_cpu():
+    pytest.skip("need x86_64-v3 capable CPU", allow_module_level=True)
+
+
+@pytest.fixture(name="build_container", scope="session")
+def build_container_fixture():
+    subprocess.check_call([
+        "podman", "build",
+        "-f", "Containerfile",
+        "-t", "bootc-image-builder-test",
+    ])
+
 
+@pytest.fixture(name="build_image", scope="session")
+def build_image_fixture(tmpdir_factory, build_container):
+    ret = collections.namedtuple(
```

Comment:
This confused me :) by the time I read the end of the function I'd forgotten about the namedtuple so I went searching for a function named `ret()`, maybe move it to the end or name it something a bit more descriptive.